src/chain.js

```diff
@@ -11,19 +11,25 @@ var map = require('./map');
  *
  * Dispatches to the `chain` method of the second argument, if present,
  * according to the [FantasyLand Chain spec](https://github.com/fantasyland/fantasy-land#chain).
+ * Treats the second argument as a FantasyLand Chain, when the the second argument is a function
+ * of the form `a -> x -> b`.
  *
  * @func
  * @memberOf R
  * @since v0.3.0
  * @category List
  * @sig Chain m => (a -> m b) -> m a -> m b
+ * @sig (x -> a) -> (a -> x -> b) -> (x -> b)
```

Comment:
> I'd prefer to let the general signature tell the story.

Does it tell the story? Or is it worth explicitly calling out the function case? At present the docs state:

```js
* Dispatches to the `chain` method of the second argument, if present,
* according to the [FantasyLand Chain spec]
```

Nobody reading the docs could tell that `Function` would work in this manner, since `Function` (typically) does not have a `chain` method.